### DISCUSSION TITLE
I found something that looks like syntax error in netrw. Is it an error or just as planned?

### DISCUSSION THREAD

**Discussion by msva** [vim/vim#15717] [Q&A]
Hi there!

I found something that looks like a typo or another kind of mistake in netrw: https://github.com/vim/vim/blob/c18a9d5835456e0e47e943b673d631caaebdbea3/runtime/autoload/netrw.vim#L4236C47-L4236C48
(notice the closing `]` without opening `[`).
But I'm not sure if it is a bug or it is intended to be as that, since no one noticed that for last 5 years.
So I'm not sure, if it is needed to be "fixed" at all, and in particular should I send a PR with removing that trailing `]`.

**Comment by zzzyxwvut** [comment]
You can simulate the logic as follows:

```vim
let s:netrwbuf= {}
enew
let s:netrwbuf["NetrwTreeListing"] = bufnr()
let bufnum= s:netrwbuf["NetrwTreeListing"]
wincmd p
exec 'bwipeout ' . bufnum

if !bufexists(bufnum)
 call remove(s:netrwbuf,"NetrwTreeListing"])
 let bufnum= -1
endif

echo bufnum
```

Save it and source it.

I see `E116`.  Possibly a copy-pasting slip.

**Comment by telemachus** [comment]
@msva Thanks for pointing this out. It should be fixed by #15718. (It's also been merged into neovim now since you asked there first. https://github.com/neovim/neovim/pull/30475)
